### PR TITLE
[VR-11876] Profile training data

### PR DESCRIPTION
**Pull request by convoliution**
## Changes
- Add `RegisteredModelVersion.log_training_data_profile()` for logging [`FeatureDataInModelVersion`](https://github.com/VertaAI/protos-all/blob/master/protos/protos/public/monitoring/DeploymentIntegration.proto)s with helper functions and tests
  - light context: this profiles each column of input and output dataframes for their missing values + their actual values (discrete/continuous) and stores info for the profilers + distributions as attributes

## Tests
```bash
$ pytest test_model_registry/test_model_version.py::TestAutoMonitoring
==================================================== test session starts ====================================================
platform darwin -- Python 3.7.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/miliu/Documents/modeldb/client/verta/tests, configfile: pytest.ini
plugins: hypothesis-6.14.0
collected 4 items                                                                                                           

test_model_registry/test_model_version.py ....                                                                        [100%]

========================================= 4 passed, 8 warnings in 63.86s (0:01:03) ==========================================
```

### CODE REVIEW

File: client/verta/verta/registry/entities/_modelversion.py [hunk 1]
```diff
@@ -94,7 +105,14 @@ def __repr__(self):
                 ),
                 "description: {}".format(msg.description),
                 "labels: {}".format(msg.labels),
-                "attributes: {}".format(_utils.unravel_key_values(msg.attributes)),
+                "attributes: {}".format(
+                    {
+                        key: val
+                        for key, val
+                        in _utils.unravel_key_values(msg.attributes).items()
+                        if not key.startswith("__verta_")
```
Comment (daniel-verta) [inline client/verta/verta/registry/entities/_modelversion.py]: optionally: name this string constant something like `INTERNAL_PREFIX`

File: client/verta/verta/registry/entities/_modelversion.py [hunk 2]
```diff
@@ -1075,3 +1093,297 @@ def delete(self):
...
+            ``"num_unique"`` and ``"type"`` for each column in `df`.
+
+        """
+        metadata = {}
+        for column in df:
+            metadata[column] = {}
+            metadata[column]["num_unique"] = df[column].value_counts().size
+            metadata[column]["type"] = str(df[column].dtypes)
+
+        return metadata
```
Comment (daniel-verta) [inline client/verta/verta/registry/entities/_modelversion.py]:
```suggestion
        return {column: metadata for column, metadata in df.apply(lambda col: {"num_unique": col.value_counts().size, "type": str(col.dtypes)}
```

Comment (daniel-verta) [reply client/verta/verta/registry/entities/_modelversion.py]: (Or something similar using `df.apply`)

**Review by daniel-verta — APPROVED** [review]
At some point (maybe soon) we'll probably want to ditch some of the looping in here in favor of using `pandas.Dataframe.apply`. This looks correct for now though and we can save optimization for later.

File: client/verta/verta/registry/entities/_modelversion.py (hunk 2 above)
Comment (convoliution) [reply client/verta/verta/registry/entities/_modelversion.py]: Done! I do remember reading that `df.apply()` is much more efficient than `for`-looping over the DataFrame.

**Pull request merged**